src/pheval/utils/utils.py

```diff
@@ -0,0 +1,143 @@
+"""Contains all pheval utility methods"""
+import logging
+import random
+from pathlib import Path
+from typing import List
+
+import pandas as pd
+
+info_log = logging.getLogger("info")
+
+
+def rand(self: pd.DataFrame, min_num: int, max_num: int, scramble_factor: float) -> float:
+    """
+    Numeric scrambling
+    Args:
+        self (pd.DataFrame): dataframe records
+        min_num (int): min value from this records
+        max_num (int): max value from this records
+        scramble_factor (float): scramble factor scalar
+    Returns:
+        float: randomized number
+    """
+    try:
+        return self + (random.uniform(min_num, max_num) * scramble_factor)  # noqa
+    except TypeError as err:
+        info_log.error(self, exc_info=err)
+        return self
+
+
+def semsim_randomisation(
+    input: Path,
+    output: Path,
+    scramble_factor: float = 0.5,
+    columns_to_be_scrambled: List[str] = tuple(["jaccard_similarity"]),
+) -> pd.DataFrame:
+    """
+    Scrambles semantic similarity profile with a magnitude between 0 and 1 (scramble_factor:
+    0 means no scrambling and 1 means complete randomisation).
+    It then randomises the above scores with a degree of the scramble_factor
+    and returns a scrambles pandas dataframe.
+        Args:
+              input (Path):
+              scramble_factor (float) scalar scramble factor
+              columns_to_be_scrambled (List[str], optional): [description].
+              Defaults to ['jaccard_similarity'].
+              output (Path)
+        Returns:
+            pd.Dataframe: scrambled dataframe
+    """
+    semsim = pd.read_csv(input, sep="\t")
+    dataframe = scramble_semsim_df(semsim, scramble_factor, columns_to_be_scrambled)
+    dataframe.to_csv(output, sep="\t", index=False)
+
+
+def scramble_semsim_df(
+    dataframe: pd.DataFrame,
+    scramble_factor: float,
+    columns_to_be_scrambled: List[str],
+) -> pd.DataFrame:
+    """scramble_semsim_df
+    Args:
+        dataframe (pd.DataFrame): [description]
+        scramble_factor (float): [description]
+        columns_to_be_scrambled (List[str]): [description]
+    Returns:
+        pd.DataFrame: [description]
+    """
+    for col in columns_to_be_scrambled:
+        min_num = dataframe[col].min()
+        max_num = dataframe[col].max()
+        dataframe[col] = dataframe[col].apply(rand, args=(min_num, max_num, scramble_factor))
+    return dataframe
+
+
+def semsimconvert(input: Path, output: Path, subject_prefix: str, object_prefix: str):
```

Comment:
create a general `semsimconvert` method with a "output_format" parameter, and move all the code in this method here to a specific: `semsimconvert_exomiserdb` or similar.

Reply:
Or to keep it like the h2 method: `semsim2exomiserdb`.

Reply:
This still looks like a method with a general name (semsimconvert) and a specific purpose (converting exomiser db tables).. Do you object renaming it?